js/services/data-service.js

```diff
@@ -52,6 +81,63 @@ export default function createDataService(assetUrls) {
       return response.data;
     },
 
+    /**
+     * Inspects chart data and collects warnings.
+     * @param {String} dateField - the name of the date field used by the chart.
+     * @param {Promise->{data: Object[], filterEvents: String[]}} responseData - REDCap record
+     *   data extracted from the request.
+     * @return {Promise->{data: Object[], filterEvents: String[], warnings: Object[]}} the
+     *   REDCap data response, with an array of warnings added if problematic data are found.
+     */
+    _validateChartData(dateField, responseData) {
+      let { data, filterEvents } = responseData;
+      data = data || [];
+      filterEvents = filterEvents || [];
+
+      let blankDates = 0;
+      let repeatingInstruments = false;
+
+      data.forEach(record => {
+        if (record[dateField] === '') {
+          blankDates++;
+        }
+
+        if (record.redcap_repeat_instance !== undefined) {
+          repeatingInstruments = true;
+        }
+      });
+
+      const warnings = [
+        data.length === 0 ? makeWarning('noData') : null,
+        filterEvents.length > 1 ? makeWarning('multipleEvents') : null,
+        blankDates > 0 ? makeWarning('blankDateFields', blankDates) : null,
+        repeatingInstruments ? makeWarning('repeatingInstruments') : null
+      ].filter(Boolean);
+
+      return {
+        data,
+        filterEvents,
+        warnings
+      };
+    },
+
```

Comment:
Filtering out records with blank date fields used to be done in multiple places, leading to regressions like the one Matt found recently. Hopefully filtering them out while processing the response will fix this for good.